monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+                label: label with shape [image_shape]
+        The data case is stored in self.data
+        """
+        self.data = {}
+        self.data["image"] = batch_data["image"].to(self.device)  # torch.tensor(batch_data['image']).to(self.device)
+        self.data["label"] = batch_data["label"].to(self.device)  # torch.tensor(batch_data['label']).to(self.device)
+        self.data["image_meta_dict"] = batch_data["image_meta_dict"]
+        self.data["label_meta_dict"] = batch_data["label_meta_dict"]
+        case_stats = {}
+        for func in self.functions:
+            case_stats.update(func())
+        return case_stats
+
+    def _get_case_summary(self):
+        """
+        Function to combine case stats. The stats for each case is stored in self.result['stats_by_cases'].
+        Each case stats is a dictionary. The function first get all the leaf-keys of self.gather_summary.
+        self.gather_summary is a dictionary of the same structure with the final summary yaml
+        output (self.results['stats_summary']), because it is updated by case_stats_summary.
+        The operations is retrived by recursive_getvalue and the combined value is calculated.
+
+        """
+        # initialize gather_summary
+        [func() for func in self.functions_summary]
+        # read
+        key_chains = recursive_getkey(self.gather_summary)
+        self.results["stats_summary"] = copy.deepcopy(self.gather_summary)
+        for key_chain in key_chains:
+            opt = recursive_getvalue(self.gather_summary, key_chain)
+            value = opt(
+                list(filter(None, [recursive_getvalue(case, key_chain) for case in self.results["stats_by_cases"]]))
+            )
+            recursive_setvalue(key_chain, value, self.results["stats_summary"])
+
+    def get_all_case_stats(self):
+        """Get all case stats. Caller of the DataAnalyser class."""
+        start = time.time()
+        self.results = {"stats_summary": {}, "stats_by_cases": []}
+        s = start
+        if not has_tqdm:
+            warnings.warn("tqdm is not installed. not displaying the caching progress.")
+
+        for batch_data in tqdm(self.dataset) if has_tqdm else self.dataset:
+            images_file = batch_data[0]["image_meta_dict"]["filename_or_obj"]
+            label_file = batch_data[0]["label_meta_dict"]["filename_or_obj"]
+            logger.debug(f"Load data spent {time.time() - s}")
+            case_stat = {"image": images_file, "label": label_file}
+            case_stat.update(self.get_case_stats(batch_data[0]))
+            self.results["stats_by_cases"].append(case_stat)
+            logger.debug(f"Process data spent {time.time() - s}")
+            s = time.time()
+        self._get_case_summary()
+        self.save_yaml(self.results)
+        logger.debug(f"total time {time.time() - start}")
+        return self.results
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="input")
```

Comment:
Do you plan to use Google fire for the CLI in this PR?

Thanks.

Reply:
Hi @Nic-Ma, I am tinkering the __main__ of the DataAnalyzer and the other future modules to make the inputs/outputs compatible with the original repo. The overall purpose is to enable an integrated test after we finish all of them.

Reply:
Hi @mingxin-zheng ,

Please add the `fire` based python APIs, refer to: https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/__main__.py

Thanks.